modules/tensor_mechanics/include/kernels/ADInertialForceShell.h

```diff
@@ -0,0 +1,186 @@
+  virtual void computeResidualForJacobian(ADDenseVector & residual);
+
+private:
+  /// Number of coupled rotational variables
+  unsigned int _nrot;
+
+  /// Number of coupled displacement variables
+  unsigned int _ndisp;
+
+  /// Variable numbers corresponding to rotational variables
+  std::vector<unsigned int> _rot_num;
+
+  /// Variable numbers corresponding to displacement variables
+  std::vector<unsigned int> _disp_num;
+
+  /// Variable numbers corresponding to velocity aux variables
+  std::vector<unsigned int> _vel_num;
+
+  /// Variable numbers corresponding to acceleraion aux variables
+  std::vector<unsigned int> _accel_num;
+
+  /// Variable numbers corresponding to rotational velocity aux variables
+  std::vector<unsigned int> _rot_vel_num;
+
+  /// Variable numbers corresponding to rotational acceleration aux variables
+  std::vector<unsigned int> _rot_accel_num;
+
+  /**
+   * Rotational transformation from global to initial beam local
+   * coordinate system
+   **/
+  ADRankTwoTensor _original_local_config;
+
+  /// Direction along which residual is calculated
+  const unsigned int _component;
+
+  /**
+   * Current translational and rotational velocities at the two nodes
+   * of the beam in the global coordinate system
+   **/
+  ADRealVectorValue _vel_0, _vel_1, _rot_vel_0, _rot_vel_1;
+  ADRealVectorValue _vel_2, _vel_3, _rot_vel_2, _rot_vel_3;
+  /**
+   * Current translational and rotational velocities at the two nodes
+   * of the beam in the global coordinate system
+   **/
+  ADRealVectorValue _old_vel_0, _old_vel_1, _old_rot_vel_0, _old_rot_vel_1;
+  ADRealVectorValue _old_vel_2, _old_vel_3, _old_rot_vel_2, _old_rot_vel_3;
+  /**
+   * Current translational and rotational accelerations at the two nodes
+   * of the beam in the global coordinate system
+   **/
+  ADRealVectorValue _accel_0, _accel_1, _rot_accel_0, _rot_accel_1;
+  ADRealVectorValue _accel_2, _accel_3, _rot_accel_2, _rot_accel_3;
+  /**
+   * Current translational and rotational velocities at the two nodes
+   * of the beam in the initial beam local coordinate system
+   **/
+  ADRealVectorValue _local_vel_0, _local_vel_1, _local_rot_vel_0, _local_rot_vel_1;
+  ADRealVectorValue _local_vel_2, _local_vel_3, _local_rot_vel_2, _local_rot_vel_3;
+  /**
+   * Current translational and rotational velocities at the two nodes
+   * of the beam in the initial beam local coordinate system
+   **/
+  ADRealVectorValue _local_old_vel_0, _local_old_vel_1, _local_old_rot_vel_0, _local_old_rot_vel_1;
+  ADRealVectorValue _local_old_vel_2, _local_old_vel_3, _local_old_rot_vel_2, _local_old_rot_vel_3;
+  /**
+   * Current translational and rotational accelerations at the two nodes
+   * of the beam in the initial beam local coordinate system
+   **/
+  ADRealVectorValue _local_accel_0, _local_accel_1, _local_rot_accel_0, _local_rot_accel_1;
+  ADRealVectorValue _local_accel_2, _local_accel_3, _local_rot_accel_2, _local_rot_accel_3;
```

Comment:
You have a ton of those sets. I would strongly suggest to make a struct that holds your 3 position and rotation vector quantities!

```
struct PosRotVectors
{
   std::array<ADRealVectorValue, 3> pos;
   std::array<ADRealVectorValue, 3> rot;
};
```

then you'd have 

```
...
/// Current velocities at the two nodes in the initial beam local coordinate system
PosRotVectors _local_vel;

/// Old velocities at the two nodes in the initial beam local coordinate system
PosRotVectors _local_old_vel;

/// Current accelerations at the two nodes in the initial beam local coordinate system
PosRotVectors _local_accel;
```

easier to read. Then `_local_rot_accel_1 -> _local_accel.rot[1]`